tasks/enable_restart_vpn.yml

```diff
@@ -0,0 +1,20 @@
+---
+- block:
+  - name: enable and restart vpn services
+    service:
+      name: "{{ item }}"
+      state: restarted
+      enabled: yes
+    async: 1
+    poll: 0
+    loop: "{{ __vpn_services }}"
+
+  - name: Wait for ssh connection to return
+    wait_for:
+      host: "{{ ansible_default_ipv4.address |\
+            d(ansible_default_ipv6.address) }}"
+      port: 22
+      delay: 3
+      state: started
+    delegate_to: localhost
+    become: false
```

Comment:
is it necessary to use `delegate_to: localhost` and `become: false` here?

Reply:
`become: false` is unnecessary. But the `delegate_to: localhost` is necessary because without it, ansible tries to establish an ssh connection to the target host before waiting for port 22 to be open. The workaround is to delegate the task to `localhost` (which will always be reachable) and put the ip address of the target host in the `host: ...` parameter of `wait_for` (this is what the task is doing right now).

Reply:
Also - why is this in a `block`?

Reply:
You're right, it doesn't need to be in a block. That was left over from an earlier implementation.